### DISCUSSION TITLE
Protected branches: status reports are being ignored with a version matrix strategy.

### DISCUSSION THREAD

**Discussion by g105b** [community/community#47870] [Pull Requests]
### Select Topic Area

Bug

### Body

Hi Githubbers!

I've recently introduced a matrix strategy in my Github Actions builds. This means that my repositories are now tested running multiple versions of the programming language. It works a treat, but now it's introduced a bug when using protected branches.

As you can see in this screenshot, I'm trying to require a status check to pass on my protected `master` branch:

![Requiring status checks on a protected branch](https://user-images.githubusercontent.com/358014/219863931-7a41faf8-d351-4a63-a318-9eb8b43d5ce4.png)

The dropdown shows "composer" (the name of one of the steps in my Github Actions Workflow file), but then it also shows the different version numbers that are currently configured.

I chose "composer", as the different versions in the matrix strategy will change for this workflow step, and the multiple other steps. I don't want to have to go and change my branch protection rules for every release of the programming language, for every step of every repository I maintain.

After creating a pull request, I can see that the two individual versioned composer runs have completed successfully, but the PR is blocked because it doesn't think the main "composer" step has finished. It's been over an hour now, and it doesn't look like it will ever be marked as complete:

![forever waiting for my status check](https://user-images.githubusercontent.com/358014/219858171-12201ddf-9d06-4f4b-95e9-c01421cc3a1b.png)

Any ideas of how I should handle this? I maintain many repositories that will often have their version matrix strategies updated, and each repository has multiple steps that behave in the same way. I want to "set and forget" my branch protection rules, but it currently doesn't look like this will be possible.

Thanks,
Greg.

P.S. The first repository I've added this rule to is https://github.com/PhpGt/Curl, because it's a really simple codebase - [the PR is here](https://github.com/PhpGt/Curl/pull/67) and the [workflow file is here](https://github.com/PhpGt/Curl/blob/master/.github/workflows/ci.yml#L6-L33).

**Comment by rogersillito** [comment]
I'm seeing the same thing: the "parent" is causing status checks on the protected branch to hang with reference the matrix "parent", even though all "child" matrix jobs have run & succeeded:

![286672339-c7a1b72f-e830-4eec-bf58-632e4631613c](https://github.com/community/community/assets/2572893/98d613c2-4143-4f50-abc9-0d3e80f4dd90)
